File: scala/functionalcolorselector/src/main/scala/ColorChangerPanel.scala
import java.awt.{Graphics, Graphics2D}
import java.awt.event.{KeyEvent, KeyListener}

import cats.effect.IO
import javax.swing.JPanel

class ColorChangerPanel extends JPanel with KeyListener {

  override def keyPressed(e: KeyEvent): Unit =
    e.getKeyCode match {
      case KeyEvent.VK_SPACE => Gui.updatePanelBackgroundColor(this).unsafeRunSync()
      case k => IO.apply(println(s"key pressed : $k"))
    }

  override def keyTyped(e: KeyEvent): Unit = ()
  override def keyReleased(e: KeyEvent): Unit = ()

}

File: kotlin/kotlinFunctionalColorSelector/src/main/kotlin/functionalcolorselector/ColorChangerPanel.kt
package functionalcolorselector

import arrow.fx.IO
import java.awt.event.KeyEvent
import java.awt.event.KeyListener

import javax.swing.JPanel

class ColorChangerPanel : JPanel(), KeyListener {

    override fun keyPressed(e: KeyEvent) {
        when (e.keyCode) {
            KeyEvent.VK_SPACE -> Gui.updatePanelBackgroundColor(this)
            else -> IO { println("Key pressed: ${e.keyCode}") }
        }.unsafeRunSync()
    }

    override fun keyTyped(e: KeyEvent) {}
    override fun keyReleased(e: KeyEvent) {}

}